File: src/enigma/general.enigma.ts
import { EncryptionMethods, type EnigmaSchema } from "@/lib/encryption";

export function GeneralEnigmaSchema(key: string): EnigmaSchema[] {
  return [
    {
      method: EncryptionMethods.AES,
      key: () => key, // Layer 1: AES with original key
    },
    {
      method: EncryptionMethods.AES,
      key: () => key.split("").reverse().join(""), // Layer 2: AES with reversed key
    },
    {
      method: EncryptionMethods.AES,
      key: () => key.slice(0, Math.floor(key.length / 2)), // Layer 3: AES with first half of the key
    },
    {
      method: EncryptionMethods.AES,
      key: () => key.slice(Math.floor(key.length / 2)), // Layer 4: AES with second half of the key
    },
    {
      method: EncryptionMethods.base64, // Layer 5: Base64 Encoding
    },
  ];
}

File: core/enigma/general.enigma.go
package enigma

import "react-go/core/function"

func GeneralEnigmaSchema(key string) []function.EnigmaSchema {
	return []function.EnigmaSchema{
		{
			Method: function.AES,
			Key:    func() string { return key }, // Layer 1: AES with original key
		},
		{
			Method: function.AES,
			Key:    func() string { return function.ReverseStrings(key) }, // Layer 2: AES with reversed key
		},
		{
			Method: function.AES,
			Key:    func() string { return key[:len(key)/2] }, // Layer 3: AES with first half of the key
		},
		{
			Method: function.AES,
			Key:    func() string { return key[len(key)/2:] }, // Layer 4: AES with second half of the key
		},
		{
			Method: function.Base64, // Layer 5: Base64 Encoding
		},
	}
}
